syntax = "proto3";

option java_package = "com.grpc.demo.service";

package donation;

message Empty{}

message Donation {
    int32 id = 1;
    int32 categoria = 2;
    string description = 3;
    int32 cantidad = 4;
    string username = 5;
    bool eliminado = 6;
 }

message DonationList {
    repeated Donation donation = 1;
}

message Response {
    bool success = 1;
    string message = 2;
}

message DonationTransferRequest {
    int32 request_id = 1;
    int32 donor_org_id = 2;
    int32 recipient_org_id = 3;
    repeated Donation donations = 4;
}

message DonationOfferRequest {
    int32 offer_id = 1;
    int32 donor_org_id = 2;
    repeated Donation donations = 3;
}

message DonationRequestCancellationRequest {
    int32 org_id = 1;
    int32 request_id = 2;
}

service DonationService {
    rpc CreateDonation(Donation) returns(Response);
    rpc UpdateDonation(Donation) returns(Response);
    rpc DeleteDonation(Donation) returns(Response);
    rpc ListDonations(Empty) returns(DonationList);
    rpc TransferDonation(DonationTransferRequest) returns(Response);
    rpc OfferDonation(DonationOfferRequest) returns(Response);
    rpc CancelDonationRequest(DonationRequestCancellationRequest) returns(Response);
}